app/assets/javascripts/covid19/src/webapp/components/structures-table/Columns.tsx

```diff
@@ -90,34 +100,30 @@ export const columnsBase: Columns = [
         renderCell: DetailsCell,
         renderString: row => {
             const { details } = row;
-            let string = "";
-            if (details == null) return "";
-            string += details?.sample?.name ?? "";
-            string += details?.sample?.macromolecules?.join(", ") ?? "";
-            string += details?.sample?.assembly ?? "";
-            string += details?.sample?.exprSystem ?? "";
-            string += details?.sample?.uniProts?.join(", ") ?? "";
-            string += details?.sample?.genes?.join(", ") ?? "";
-            string += details?.sample?.bioFunction?.join(", ") ?? "";
-            string += details?.sample?.bioProcess?.join(", ") ?? "";
-            string += details?.sample?.cellComponent?.join(", ") ?? "";
-            string += details?.sample?.domains?.join(", ") ?? "";
-            string +=
-                details?.refdoc
-                    ?.map(
-                        ref =>
-                            ref.id +
-                            ", " +
-                            ref.title +
-                            ", " +
-                            ref.authors +
-                            ", " +
-                            ref.journal +
-                            ", " +
-                            ref.abstract
-                    )
-                    .join(", ") ?? "";
-            return string;
+            if (!details) return "";
+            return _.compact(
+                [
+                    details?.sample?.name,
+                    details?.sample?.macromolecules,
+                    details?.sample?.assembly,
+                    details?.sample?.exprSystem,
+                    details?.sample?.uniProts,
+                    details?.sample?.genes,
+                    details?.sample?.bioFunction,
+                    details?.sample?.bioProcess,
+                    details?.sample?.cellComponent,
+                    details?.sample?.domains,
+                    details?.refdoc?.map(ref => [
+                        ref.id,
+                        ref.idLink,
+                        ref.title,
+                        ref.authors,
+                        ref.journal,
+                        ref.pubDate,
+                        ref.doi,
+                    ]),
+                ].flat()
+            ).join(",");
```

Comment:
I know it's a pain, but it's preferable to have the final value before the join to be `string[]`. If we have an heterogeneous data input, we can end up with a weird joining. Proposal: use `...` when needed and helpers to make the code look nicer (and flatMap for refdoc):

```
            const { sample } = details;
            const array = (values: string[] | undefined) => values || [];

            const values: string[] = _.compact([
                sample?.name,
                ...array(sample?.macromolecules),
                sample?.assembly,
                 // ... other fields
            ])

```